paimon-core/src/main/java/org/apache/paimon/operation/KeyValueFileStoreWrite.java

```diff
@@ -356,13 +361,26 @@ private <T> LookupLevels<T> createLookupLevels(
                         cacheManager,
                         new RowCompactedSerializer(keyType).createSliceComparator());
         Options options = this.options.toConfiguration();
+        InternalRow.FieldGetter[] getters = partitionType.fieldGetters();
+        StringBuilder builder = new StringBuilder();
+        for (InternalRow.FieldGetter getter : getters) {
```

Comment:
Extract a common method, maybe a static method in `InternalRowPartitionComputer`.